File: src/main/java/org/oewntk/yaml/in/VerbFrameProcessor.java
/*
 * Copyright (c) $originalComment.match("Copyright \(c\) (\d+)", 1, "-")2021. Bernard Bou.
 */

package org.oewntk.yaml.in;

import org.oewntk.model.VerbFrame;

import java.io.File;
import java.util.Map;

public class VerbFrameProcessor extends YamProcessor1<VerbFrame, String, String>
{
	private static final boolean DUMP = false;

	public VerbFrameProcessor(final File dir)
	{
		super(dir);
		this.dir = dir;
	}

	@Override
	protected File[] getFiles()
	{
		return dir.listFiles((f) -> f.getName().matches("frames.yaml"));
	}

	@Override
	protected VerbFrame processEntry(String source, Map.Entry<String, String> entry)
	{
		String id = entry.getKey();
		String v = entry.getValue();
		if (DUMP)
		{
			Tracing.psInfo.println(id);
		}
		return new VerbFrame(id, v);
	}
}

File: src/main/java/org/oewntk/yaml/in/VerbFrameProcessor.kt
/*
 * Copyright (c) 2024. Bernard Bou.
 */
package org.oewntk.yaml.`in`

import org.oewntk.model.SenseKey
import org.oewntk.model.VerbFrame
import org.oewntk.model.VerbFrameType
import java.io.File

/**
 * Verb frames processor
 *
 * @param dir dir containing YAML files
 */
class VerbFrameProcessor(dir: File) : YamProcessor1<VerbFrame, String, String>(dir) {

    init {
        this.dir = dir
    }

    override val files: Array<File>
        get() = dir.listFiles { f: File -> f.name.matches("frames.yaml".toRegex()) }!!

    override fun processEntry(source: String?, entry: Pair<SenseKey, VerbFrameType>): VerbFrame {
        val id = entry.first
        val v = entry.second
        if (DUMP) {
            Tracing.psInfo.println(id)
        }
        return VerbFrame(id, v)
    }

    companion object {

        private const val DUMP = false
    }
}
